### DISCUSSION TITLE
Upgrade Netbox v2.11.4 to v2.11.5 or later error

### DISCUSSION THREAD

**Discussion by thoms27** [netbox-community/netbox#6558] [Q&A] [answered]
### NetBox version

v2.11.4

### Python version

3.6

### Steps to Reproduce

1. Execute ./upgrade.sh script

### Expected Behavior

No error

### Observed Behavior

./upgrade.sh
Removing old virtual environment...
Creating a new virtual environment at /opt/netbox/apps/venv...
Error: Command '['/opt/netbox/apps/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.

  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib64/python3.6/ensurepip/__main__.py", line 5, in <module>
    sys.exit(ensurepip._main())
  File "/usr/lib64/python3.6/ensurepip/__init__.py", line 232, in _main
    default_pip=args.default_pip,
  File "/usr/lib64/python3.6/ensurepip/__init__.py", line 103, in _bootstrap
    dep_records = rewheel.find_system_records([p[0] for p in _PROJECTS])
  File "/usr/lib64/python3.6/ensurepip/rewheel/__init__.py", line 39, in find_system_records
    sys_sitepack = site.getsitepackages([sys.base_prefix, sys.base_exec_prefix])
AttributeError: module 'site' has no attribute 'getsitepackages'

**Comment by thoms27** [comment, marked as answer]
Until now, my procedure to upgrade worked. Why is do not work anymore ?

**Reply by candlerb** [reply]
Were you inside a virtualenv at the time you tried that? The prompt normally says "(venv)" if you are.

If so, try exiting the virtualenv with "deactivate" before running `./upgrade.sh` - or just logout and login again.

**Reply by thnow** [reply]
I've the same effect, updated from 2.11.4 to 2.11.5 and get "Error: Command '['/opt/netbox/apps/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1." 
OS ist Centos 7 and Python 3.6.8 and all Updates ran fine before. I'm not in any venv at that moment and 100% following the upgrade guide. I can reproduce it at any time, if more information is needed.

**Reply by jeremystretch** [reply]
@thnow what is the full error that you get? Is it the same as above (`AttributeError: module 'site' has no attribute 'getsitepackages'`)?

Also, what version of pip do you have installed? (`pip -V`)

**Reply by thnow** [reply]
While checking what you requested I discovered that someone has mangled the python version on this system, it was a 3.5 pip. I fixed that, for now and the upgrade script worked so far, now I have another error while migrating, but I will look into that separately.

**Reply by thnow** [reply]
To sum it up, it was my fault at all, my pip was to old, after fixing that, I ran into the `ValueError: Cannot query "VRF object (1)": Must be "VRF" instance.` error witch is fixed in 2.11.6. So I skipped 2.11.5 and I'm now on the latest release as I like to be. Thank your for providing the community with such a great software!

**Reply by jeremystretch** [reply]
Awesome, glad you got it sorted!
